pinot-common/src/main/java/com/linkedin/pinot/common/config/RealtimeTagConfig.java

```diff
@@ -31,9 +29,17 @@
   public RealtimeTagConfig(TableConfig tableConfig) {
     super(tableConfig);
 
-    // TODO: after we introduce config to override tags, pick the right ones from config for consuming and completed
-    _consumingRealtimeServerTag = TenantNameBuilder.getRealtimeTenantNameForTenant(_serverTenant);
-    _completedRealtimeServerTag = TenantNameBuilder.getRealtimeTenantNameForTenant(_serverTenant);
+    TagOverrideConfig tagOverrideConfig = tableConfig.getTenantConfig().getTagOverrideConfig();
+    if (tagOverrideConfig != null && tagOverrideConfig.getRealtimeConsuming() != null) {
```

Comment:
shouldn't this be something like get(value, defaultValue) which returns defaultValue if the value is null?

Reply:
The tag override config is the config holder. This class is the one that converts it to how we want it to behave if the config is not present. I would prefer that config classes return what exactly is in the config.